File: backend/src/main/java/com/example/HopitalPlanningProject/model/AssocieId.java
package com.example.HopitalPlanningProject.model;

import jakarta.persistence.Embeddable;
import lombok.*;

import java.io.Serializable;
import java.util.Objects;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class AssocieId implements Serializable {
    private int idShift;
    private int idMotif;

    @Override
    public int hashCode() {
        return Objects.hash(idShift, idMotif);
    }
}

File: frontend/src/models/AssocieId.js
// models/AssocieId.js
class AssocieId {
    constructor(idShift, idMotif) {
        this.idShift = idShift;
        this.idMotif = idMotif;
    }

    equals(other) {
        return other instanceof AssocieId &&
               this.idShift === other.idShift &&
               this.idMotif === other.idMotif;
    }

    hashCode() {
        return `${this.idShift}-${this.idMotif}`;
    }
}

export default AssocieId;
